src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java

```diff
@@ -127,7 +132,12 @@ static User findByUsername(String username) {
 
             // Some mods store other files in the 'playerdata' folder, so
             // we need to ensure that the filename is a valid UUID
-            if (playerUuid.split("-").length != 5) {
+            //
+            // It's also important to note that the SaveHandler's
+            // getAvailablePlayerDat method will only remove the
+            // extension from ".dat" files, any other file will retain
+            // their extension.
+            if (!UUID_PATTERN.reset(playerUuid).matches()) {
```

Comment:
Fair point. I was thinking about performance, but didn't think about people doing this async. I'll create the matcher within the method, but outside the for loop.